### PR TITLE
refactor: use external protocols & update to iroh@0.29

### PR DESCRIPTION
**Pull request by dignifiedquire**
## uniffi

- [x] remove protocols from `iroh`
- [x] switch client `iroh-gossip`
- [x] switch client `iroh-docs`
- [x] switch client `iroh-blobs`
- [x] remove remote client 
- [x] update to 0.29
- [x] update tests
- [x] udpate config options
- [x] use gradle to build and test kotlin

## js

- [x] remove protocols from `iroh`
- [x] switch client `iroh-gossip`
- [x] switch client `iroh-docs`
- [x] switch client `iroh-blobs`
- [x] remove remote client 
- [x] update tests
- [x] udpate config options

### CODE REVIEW

File: src/blob.rs [hunk 1]
```diff
@@ -6,29 +6,54 @@ use std::{
...
+    client: MemClient,
+    net_client: iroh::client::net::Client,
 }
 
 #[uniffi::export]
 impl Iroh {
     /// Access to blob specific funtionaliy.
     pub fn blobs(&self) -> Blobs {
-        Blobs { node: self.clone() }
+        let client = match self.storage {
```
Comment (rklaehn) [inline src/blob.rs:27]: This is a bit painful...

**Comment by Arqu** [comment]
FYI bumped versions for manylinux which should resolve the current CI issues. As per https://github.com/pypa/manylinux manylinux_2014 should be very close to EoL even though it's still supported by the project.

File: kotlin/lib/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
+
+    // Use the JUnit 5 integration.
+    testImplementation(libs.junit.jupiter.engine)
+
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+
+    implementation("net.java.dev.jna:jna:5.15.0")
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.9.0")
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-test:1.9.0")
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply kotlin/lib/build.gradle.kts]: fixed

File: kotlin/computer/iroh/iroh_ffi.kt
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply kotlin/computer/iroh/iroh_ffi.kt:1]: yep, removed

File: iroh-js/src/gossip.rs [hunk 3]
```diff
@@ -75,21 +76,19 @@ impl From<SubscribeResponse> for Message {
...
-    }
-}
-
-impl Gossip {
-    fn client(&self) -> &iroh::client::Iroh {
-        self.node.inner_client()
+        let gossip = self
+            .router
+            .get_protocol(iroh_gossip::net::GOSSIP_ALPN)
+            .expect("no gossip available");
```
Comment (ramfox) [inline iroh-js/src/gossip.rs:90]: Just calling out that we are removing `get_protocol` in the next release.

Comment (ramfox) [reply iroh-js/src/gossip.rs:90]: ...i guess this actually doesn't matter since we are removing the protocols in ffi next release anyway.

**Review by ramfox — APPROVED** [review]
Looks good!

**Pull request merged**